scripts/globals/dark_ixion.lua

```diff
@@ -0,0 +1,471 @@
+    then
+        -- leave zone alone, push back repop ... since zone was already randomized, implied by PopTime being in the future
+        -- 'If he was not due to spawn during this time frame (after maintenance), his spawn window will reset to 21 hours after servers come online.'
+        SetServerVariable('DarkIxion_PopTime', os.time() + 21 * 60 * 60)
+    end
+end
+
+xi.darkixion.zoneOnGameHour = function(zone)
+    local ixionID = xi.darkixion.zoneinfo[zone:getID()].mobID
+    local ixion = GetMobByID(ixionID)
+    if
+        GetServerVariable('DarkIxion_ZoneID') == zone:getID() and
+        GetServerVariable('DarkIxion_PopTime') < os.time() - 24 * 60 * 60
+    then
+        -- wander logic in onGameHour so even sleeping zones with no players can hold DI and cycle him out
+        xi.darkixion.repop(ixion)
+    elseif
+        not ixion:isSpawned() and
+        GetServerVariable('DarkIxion_ZoneID') == zone:getID() and
+        GetServerVariable('DarkIxion_PopTime') < os.time() - 45
+    then
+        -- if gamehour flip is within 45s, randomly spawn within next twice that
+        ixion:setRespawnTime(math.random(0, 90))
+    elseif
+        ixion:isSpawned() and
+        GetServerVariable('DarkIxion_ZoneID') ~= zone:getID()
+    then
+        -- really shouldn't be possible, but catch just in case a GM manually spawned him somewhere else
+        if ixion:isEngaged() then
+            ixion:disengage()
+        else
+            DespawnMob(ixionID)
+        end
+    end
+end
+
+xi.darkixion.onMobDeath = function(mob, player, isKiller)
+    player:addTitle(xi.title.IXION_HORNBREAKER)
+    -- only reset hp after being killed
+    SetServerVariable('DarkIxion_HP', 0)
+    SetServerVariable('DarkIxion_HornStatus', 0)
+end
+
+xi.darkixion.onMobDespawn = function(mob)
+    mob:removeListener('IXION_TAKE_DAMAGE')
+    DisallowRespawn(mob:getID(), true)
+    if mob:getZoneID() == GetServerVariable('DarkIxion_ZoneID') then
+        xi.darkixion.repop(mob)
+        SetServerVariable('DarkIxion_PopTime', os.time() + math.random(20, 24) * 60 * 60) -- repop 20-24 hours after death
+    end
+end
+
+xi.darkixion.onCriticalHit = function(mob, attacker)
+end
+
+xi.darkixion.onWeaponskillHit = function(mob, attacker, weaponskill)
+end
+
+xi.darkixion.onMobInitialize = function(mob)
+end
+
+xi.darkixion.onMobSpawn = function(mob)
+    xi.darkixion.roamingMods(mob)
+    SetServerVariable('DarkIxion_PopTime', os.time())
+    mob:setLocalVar('wasKilled', 0)
+    mob:setMod(xi.mod.SLEEPRES, 100)
+    mob:setMod(xi.mod.STUNRES, 100)
+
+    mob:setMobMod(xi.mobMod.NO_REST, 10)
+    mob:setAggressive(true)
+end
+
+xi.darkixion.onMobWeaponSkillPrepare = function(mob, target)
+end
+
+xi.darkixion.onMobWeaponSkill = function(target, mob, skill)
+end
+
+xi.darkixion.onMobSkillFinished = function(mob, target, skill)
+end
+
+xi.darkixion.onAdditionalEffect = function(mob, target, damage)
+end
+
+xi.darkixion.onMobRoamAction = function(mob)
+end
```

Comment:
Same with these: Will these be needed later? If not ,they don't need to exist

Reply:
Yea. I basically wanted to build the framework for the combat portion so next PR is just filling in those functions

Reply:
I'll remove them for this PR since I don't have to touch all the files again using your helper function above